libvast/vast/concept/support/unused_type.hpp

```diff
@@ -54,6 +54,10 @@ struct unused_type : detail::equality_comparable<unused_type>,
   const unused_type& operator/=(T&&) const {
     return *this;
   }
+
+  [[deprecated]] friend inline bool operator==(unused_type, unused_type) {
```

Comment:
I'm not sure we should deprecate this unless we're actually planning to remove the remaining uses from the code base.

Reply:
That was leftover from testing, it wasn't supposed to be in here. I've removed it.